### PR TITLE
[Core] Adding Create method to constitutive_law.h

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: kratos/includes/constitutive_law.h [hunk 1]
```diff
@@ -509,6 +510,13 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
      */
     virtual ConstitutiveLaw::Pointer Clone() const;
 
+    /**
+     * creates a new constitutive law pointer
+     * @param NewId the ID of the new element
```
Comment (philbucher) [inline kratos/includes/constitutive_law.h]: copy paste error

File: kratos/includes/constitutive_law.h [hunk 2]
```diff
@@ -509,6 +510,13 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
...
+     * @return a Pointer to the new constitutive law
+     */
+    virtual Pointer Create(Kratos::Parameters& NewParameters) const;
```
Comment (philbucher) [inline kratos/includes/constitutive_law.h]: should be passed by value (@RiccardoRossi )

**Comment by philbucher** [comment]
You could also expose this method to python ...?

**Comment by loumalouomega** [comment]
I can

**Comment by loumalouomega** [comment]
Done

**Comment by loumalouomega** [comment]
Ping

**Comment by philbucher** [comment]
was this discussed with @KratosMultiphysics/technical-committee ?

**Comment by loumalouomega** [comment]
Yes, @RiccardoRossi told me to add it

**Review by philbucher — APPROVED** [review]
Approving since @RiccardoRossi initiated this

**Comment by philbucher** [comment]
btw will you also add a `Create` method that works the same way as for the Elements?

Or is this not intended?

**Comment by philbucher** [comment]
you could also add a test, there is a file for the claw-c-tests already

**Comment by loumalouomega** [comment]
I am not sure what to check

**Comment by philbucher** [comment]
Hm good question

The only thing I could imagine would be to check the type
like [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/geometrical_object.h#L148-L150)

Your call though, I think this will anyway be used in the tests in the future

**Comment by loumalouomega** [comment]
I think I will merge, we add later the test, @AlejandroCornejo needs this

**Pull request merged**